cpp/ql/lib/semmle/code/cpp/ir/implementation/aliased_ssa/Operand.qll

```diff
@@ -46,7 +46,7 @@ class Operand extends TStageOperand {
   /**
    * Gets the location of the source code for this operand.
    */
-  final Language::Location getLocation() { result = this.getUse().getLocation() }
+  final Language::Location getLocation() { result = this.getAnyDef().getLocation() }
```

Comment:
I'm not sure if it's appropriate to change _all_ the operand locations to be the location of their definition instead of their use.

For most operands, it probably doesn't really matter. I'm a bit concerned with what this will do to operands whose definition is a `PhiInstruction`. For instance, consider this example:

```cpp
void test(bool b) {
  int x;
  if(b) {
    x = 0;
  } else {
    x = 1;
  }
  // <-- there will be a `PhiInstruction` here.
  int unrelated_variable;
  int y = x;
}
```
I _think_ the location of `x` in `int y = x` will be on the same line as `int unrelated_variable_here`.

Reply:
yep, I'm analyzing now a query spitting out all operands in ir.cpp that have their use location different than their definition location, and will have a look at those. I can see for example that AddressOperand indeed would be better off with the definition location, as in something like
```
int x = 1;
```
the use points at `1`, while the definition points at `x`. But I wonder what happens to pointer dereferences? I'll dig further.

Reply:
after looking carefully at all operands where this would have made a difference, I've come to the conclusion that this should be applied to almost all RegisterOperands only. The notable exception is ThisArgumentOperand: usually both the locations are not super helpful, in some cases the def location was slightly more meaningful, but in some cases it was just absent (pointing to file://0:0:0:0), so I kept the use location there.